### DISCUSSION TITLE
Use nix binary in postUpgradeTasks

### DISCUSSION THREAD

**Discussion by chasecaleb** [renovatebot/renovate#24426] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.74.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

`nix` should be in `$PATH` of the "full" Docker image so that `postUpgradeTasks` can use it, just like other binary tools.

Use case: run `nix run 'nixpkgs#nix-prefetch-github' -- kubernetes kubernetes --rev v1.25.14 | jq --exit-status --raw-output '.hash'` as part of a `postUpgradeTasks` script so that I can update the `sha256` attribute of a package fetched with `pkgs.fetchFromGitHub`.

### What you tried so far.

`which nix` and similar in a `postUpgradeTasks` script show that `nix` is not in `$PATH` of the Docker image, even in the "full" renovate image such as `renovate/renovate:36.74.0-full@sha256:0fe099e2d08ece6fe0b2e9da9dcd3cb700bd8fa5991bbde55dbf1913bf069120`. When digging through renovate's source, it looks like it does `install-tools nix 2.10.0` to install it. I could do similar in my script... but then I need to manage updating _that_ version too.

P.S. The latest release of nix is `2.17.0`, so maybe renovate should update that :wink:

**Comment by rarkins** [comment, marked as answer]
Where do you find nix 2.10.0?

I would expect it needs to be added to https://github.com/renovatebot/docker-renovate-full/blob/main/Dockerfile

**Reply by chasecaleb** [reply]
Found it here in a test: https://github.com/renovatebot/renovate/blob/0b50feb90cc8a35ef4db4865a4d2fd6f4349f97b/lib/modules/manager/nix/artifacts.spec.ts#L225 . On looking closer though I think that has to do with the test snapshots so nevermind.

I can make a PR for https://github.com/renovatebot/docker-renovate-full/ to add Nix if that sounds reasonable to you

**Reply by chasecaleb** [reply]
For the sake of posterity: https://github.com/renovatebot/docker-renovate-full/pull/2301
